### PR TITLE
Add metrics for Scout sessions

### PR DESCRIPTION
**Pull request by brunokoeferli**
add metrics for following types of Scout session:
* client
* ui
* server
* http

359810

### CODE REVIEW

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/session/SessionMetricsHelper.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ */
+@ApplicationScoped
+public class SessionMetricsHelper {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SessionMetricsHelper.class);
+
+  protected Map<String, SessionMetrics> m_metrics = new ConcurrentHashMap<>();
+
+  public void sessionCreated(String type) {
+    Assertions.assertNotNullOrEmpty(type);
```
Comment (paolobazzi) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/session/SessionMetricsHelper.java]: use static imports here?

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/session/SessionMetricsHelper.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    LOG.info("Init session metrics of type '{}'", type);
+    Meter meter = GlobalOpenTelemetry.get().getMeter(getClass().getName());
+    return new SessionMetrics(meter, type);
+  }
+
+  protected static class SessionMetrics {
+
+    protected static final AttributeKey<String> TYPE = AttributeKey.stringKey("type");
+
+    protected Attributes m_defaultAttributes;
```
Comment (paolobazzi) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/session/SessionMetricsHelper.java]: Members could be marked final

**Review by paolobazzi — APPROVED** [review]
Looks good, see comments for minor inputs.

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/session/SessionMetricsHelper.java (hunk 1 above)
Comment (brunokoeferli) [reply org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/session/SessionMetricsHelper.java]: done

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/session/SessionMetricsHelper.java (hunk 2 above)
Comment (brunokoeferli) [reply org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/session/SessionMetricsHelper.java]: changed members to final

**Pull request merged**